Subject: b43: Change firmware missing message to refer to openSUSE script
From: Larry Finger <Larry.Finger@lwfinger.net>
Patch-mainline: never, openSUSE-specific
References: none

The error message output by b43 contains instructions for obtaining
firmware; however, this naturally does not take account of the script
/usr/sbin/install_bcm43xx_firmware. Modify the messages to suggest use
of the script.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
Modified-by: Jiri Benc <jbenc@suse.cz>
Signed-off-by: Jiri Benc <jbenc@suse.cz>

---
 drivers/net/wireless/broadcom/b43/main.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/net/wireless/broadcom/b43/main.c
+++ b/drivers/net/wireless/broadcom/b43/main.c
@@ -2165,10 +2165,12 @@ static void b43_release_firmware(struct b43_wldev *dev)
 static void b43_print_fw_helptext(struct b43_wl *wl, bool error)
 {
 	const char text[] =
-		"You must go to " \
-		"https://wireless.docs.kernel.org/en/latest/en/users/drivers/b43/developers.html#list-of-firmware " \
-		"and download the correct firmware for this driver version. " \
-		"Please carefully read all instructions on this website.\n";
+		"Please open a terminal and enter the command " \
+		"\"sudo /usr/sbin/install_bcm43xx_firmware\" to download " \
+		"the correct firmware for this driver version. " \
+		"For an off-line installation, go to " \
+		"https://en.opensuse.org/HCL:Network_(Wireless)#Broadcom" \
+		" and follow the instructions above the table.\n";
 
 	if (error)
 		b43err(wl, text);
